src/robotFactory.js

```diff
@@ -1,10 +1,64 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.coords = {};
+    this.coords.x = coords.x || 0;
+    this.coords.y = coords.y || 0;
```

Comment:
You can shorten this three assignments:
```suggestion
    this.coords = {
      x: coords.x || 0,
      y: coords.y || 0,
    };
```